storage_devices/lvm.py

```diff
@@ -498,22 +504,40 @@ def remove_pv(pv_name: str):
         cmd = f"pvremove {pv_name}"
         return TestRun.executor.run(cmd)
 
+    @staticmethod
+    def get_os_vg():
+        os_vgs = []
+        disks = get_system_disks()
+        for disk in disks:
+            cmd = (rf"pvdisplay --select pv_name=~\/dev\/{disk} | "
+                   rf"grep 'VG Name' | awk '{{print $3}}'")
+            os_vg_name = TestRun.executor.run(cmd).stdout
+            if os_vg_name:
+                os_vgs.append(os_vg_name)
+        return os_vgs
+
     @classmethod
     def remove_all(cls):
+        os_vg_names = Lvm.get_os_vg()
         cmd = f"lvdisplay | grep 'LV Path' | awk '{{print $3}}'"
         lvm_paths = TestRun.executor.run(cmd).stdout.splitlines()
         for lvm_path in lvm_paths:
             lv_name = lvm_path.split('/')[-1]
             vg_name = lvm_path.split('/')[-2]
-            cls.remove(lv_name, vg_name)
+            if vg_name not in os_vg_names:
+                cls.remove(lv_name, vg_name)
```

Comment:
having a bit of fun with this - not even a suggestion, just `-c` is my new best friend for `*display` commands ;)
```python
        cmd = "lvdisplay -c | awk -F'[/:]' '{{print $4,$5}}'"  # prints vl_name vg_name
        lvs = [tuple(lv.split(' ')) for lv in TestRun.executor.run(cmd).stdout.splitlines()]
        [cls.remove(*lv) for lv in lvs if lv[1] not in os_vg_names]]
```